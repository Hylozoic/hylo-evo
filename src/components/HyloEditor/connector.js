import { connect } from 'react-redux'
import { findMentions, clearMentions, getMentionResults } from './store'

export function mapStateToProps (state, props) {
  return {
    mentionResults: getMentionResults(state, props)
  }
}

export const mapDispatchToProps = {
  findMentions,
  clearMentions
}

export default connect(mapStateToProps, mapDispatchToProps)
