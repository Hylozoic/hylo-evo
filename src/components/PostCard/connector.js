import { connect } from 'react-redux'
import { push } from 'react-router-redux'

export function mapStateToProps (state, props) {
  return {}
}

export const mapDispatchToProps = {
  navigate: push
}

export default connect(mapStateToProps, mapDispatchToProps)
