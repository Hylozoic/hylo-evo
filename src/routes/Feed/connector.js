import { connect } from 'react-redux'
import { SAMPLE_FEED_ITEMS } from './sampleData'

function mapStateToProps (state) {
  return {
    feedItems: SAMPLE_FEED_ITEMS
  }
}

function mapDispatchToProps (dispatch, props) {
  return {}
}

export default connect(mapStateToProps, mapDispatchToProps)
