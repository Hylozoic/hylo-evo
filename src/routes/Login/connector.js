import { connect } from 'react-redux'
import { checkLogin, login } from './actions'
export default connect(null, {checkLogin, login})
