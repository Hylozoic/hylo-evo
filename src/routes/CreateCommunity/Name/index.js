import component from './Name'
import connector from './Name.connector'
export default connector(component)
