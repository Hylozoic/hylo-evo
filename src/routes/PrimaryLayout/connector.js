import { connect } from 'react-redux'
import { SAMPLE_COMMUNITY } from 'routes/Feed/sampleData'
import { fetchCurrentUser } from './actions'

function mapStateToProps ({currentUser}) {
  return {
    community: SAMPLE_COMMUNITY,
    currentUser
  }
}

export default connect(mapStateToProps, {fetchCurrentUser})
