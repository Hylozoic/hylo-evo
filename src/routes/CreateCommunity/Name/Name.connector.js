import { connect } from 'react-redux'
import { push } from 'react-router-redux'
import { addCommunityName } from '../CreateCommunity.store'
import { get } from 'lodash/fp'

export function mapStateToProps (state, props) {
  return {
    communityName: get('name', state.CreateCommunity)
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    goToNextStep: () => dispatch(push('/create-community/domain')),
    goHome: () => dispatch(push('/')),
    addCommunityName: (name) => dispatch(addCommunityName(name))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
