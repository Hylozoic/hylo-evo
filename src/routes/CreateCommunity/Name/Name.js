import React, { Component } from 'react'
import '../CreateCommunity.scss'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground, happyAxolotl } from 'utils/assets'
import { bgImageStyle } from 'utils/index'
import ModalFooter from 'components/ModalFooter'
import ModalSidebar from 'components/ModalSidebar'

export default class Name extends Component {
  constructor (props) {
    super(props)
    this.state = {
      communityName: ''
    }
  }

  handleNameChange = (event) => {
    const communityName = event.target.value
    this.setState({
      communityName
    })
  }

  submit = () => {
    this.props.addCommunityName(this.state.communityName)
    this.props.goToNextStep()
  }

  errorCheckAndSubmit = () => {
    if (this.state.communityName === '') {
      this.setState({
        error: 'Please enter a community name'
      })
    } else {
      this.submit()
    }
  }

  onEnter = event => {
    if (event.key === 'Enter') {
      this.errorCheckAndSubmit()
    }
  }

  componentWillMount = () => {
    const { communityName } = this.props
    if (communityName) this.setState({communityName})
  }

  componentDidMount = () => {
    this.props.addNetworkId()
  }

  render () {
    return <div styleName='flex-wrapper'>
      <ModalSidebar
        imageDialogOne={"Hi, I'm Hylo the Axolotl!"}
        imageDialogTwo={"It's great to meet you!"}
        imageUrl={happyAxolotl}
        onClick={this.props.goHome}
        theme={sidebarTheme}
        header="Great, let's get started"
        body="All good things start somewhere! Let's kick things off with a catchy name for your community."
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 1/3</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically'>
          <span styleName='text-input-label'>{ this.state.communityName && "What's the name of your community?"}</span>
          <TextInput
            type='text'
            name='community-name'
            onChange={this.handleNameChange}
            value={this.state.communityName}
            theme={inputTheme}
            placeholder="What's the name of your community?"
            noClearButton
            onEnter={this.onEnter}
          />
          { this.state.error && <span styleName='arrow-up' /> }
          { this.state.error && <span styleName='error'>{this.state.error}</span>}
        </div>
      </div>
      <ModalFooter
        submit={this.errorCheckAndSubmit}
        showPrevious={false}
        continueText={'Continue'}
        />
    </div>
  }
}

const inputTheme = {
  inputStyle: 'modal-input',
  wrapperStyle: 'center'
}

const sidebarTheme = {
  sidebarHeader: 'sidebar-header-full-page',
  sidebarText: 'gray-text sidebar-text-full-page'
}
