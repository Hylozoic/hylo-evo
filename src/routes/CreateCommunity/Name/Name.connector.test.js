import { mapDispatchToProps, mapStateToProps } from './Name.connector'

const dispatch = jest.fn(x => x)
const props = {}
const dispatchProps = mapDispatchToProps(dispatch, props)

describe('Domain', () => {
  it('should call fetchCommunity from mapDispatchToProps', () => {
    const communityName = 'Community Name'
    expect(dispatchProps.addCommunityName(communityName)).toMatchSnapshot()
  })

  it('should call goToNextStep from mapDispatchToProps', () => {
    expect(dispatchProps.goToNextStep()).toMatchSnapshot()
  })

  it('should call goHome from mapDispatchToProps', () => {
    expect(dispatchProps.goHome()).toMatchSnapshot()
  })

  it('should have communityName in mapStateToProps', () => {
    const communityName = 'communityName'
    const state = {
      CreateCommunity: {
        name: communityName
      }
    }
    expect(mapStateToProps(state, props).communityName).toBe(communityName)
  })
})
