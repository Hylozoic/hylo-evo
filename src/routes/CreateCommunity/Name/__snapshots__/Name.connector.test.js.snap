// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Domain should call fetchCommunity from mapDispatchToProps 1`] = `
Object {
  "payload": "Community Name",
  "type": "CreateCommunity/ADD_COMMUNITY_NAME",
}
`;

exports[`Domain should call goHome from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Domain should call goToNextStep from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/create-community/domain",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
